hudi-common/src/main/avro/HoodieClusteringStrategy.avsc

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+{
+   "namespace":"org.apache.hudi.avro.model",
```

Comment:
At a high level, don't think we need a clustering strategy avsc for this, we can just drive this from Reflection and classnames, just like CompactionPlans ?

Reply:
There needs to be coordination between scheduling and executing clustering. For example, when scheduling, we specify parameters such as 'sortColumns', 'targetFileSize' etc. These needs to be passed to async job that clusters data. These parameters are serialized and stored in the strategy/plan.

Reply:
The parameters such as 'sortColumns', 'targetFileSize' etc serialized to the clustering job makes sense to me, my question is around the strategy class name itself being written to the file. Is it just for tracking the strategy or will the async clustering job actually make use of this strategy to read our the strategyParams ?

Reply:
Yes, async strategy job would read the strategy to instantiate the class. That strategy object will be used to write clustered records (Example: strategy may use custom partitioner to respect sortColumns)